solutions/nage1234/ProducerConsumerUsingWaitNotify/producerconsumer/InventoryManagement.java

```diff
@@ -0,0 +1,37 @@
+package solutions.nage1234.ProducerConsumerUsingWaitNotify.producerconsumer;
+public class InventoryManagement {
+	private int producedItems;
+	private int capacity = 10;
+	public int getProducedItems() {
+		return producedItems;
+	}
+	
+	public synchronized void produceItem() throws Exception{
+		while(capacity >= producedItems) {
+			producedItems++;
+			Thread.sleep(500);
+			System.out.println("Produced Item = "+producedItems);
+			if(capacity == producedItems) {
+				this.notifyAll();
+				System.out.println("Producer is Waiting");			
+				this.wait();
+			}
+		}
+	}
+	
+	public synchronized void consumeItem() throws Exception{
+		int count = 0;
```

Comment:
Sorry, can you please clarify this? why is this extra Consumer specific count needed? Is this an attempt to balance out the execution between the 2 consumer threads.